ts/parts/Tooltip.ts

```diff
@@ -1188,202 +1189,281 @@ H.Tooltip.prototype = {
         labels: (string|Array<(boolean|string)>),
         points: Array<Highcharts.Point>
     ): void {
-        var tooltip = this,
-            boxes = [] as Array<Highcharts.Dictionary<any>>,
-            chart = this.chart,
-            ren = chart.renderer,
-            rightAligned = true,
-            options = this.options,
-            headerHeight = 0,
-            headerTop: boolean,
-            tooltipLabel = this.getLabel(),
-            distributionBoxTop = chart.plotTop;
+        const tooltip = this;
+        const {
+            chart: {
+                chartWidth,
+                chartHeight,
+                plotHeight,
+                plotLeft,
+                plotTop,
+                plotWidth,
+                pointer,
+                renderer: ren,
+                scrollablePixelsX = 0,
+                scrollablePixelsY = 0,
+                scrollingContainer: {
+                    scrollLeft,
+                    scrollTop
+                } = { scrollLeft: 0, scrollTop: 0 },
+                styledMode,
+                xAxis: [{ opposite }]
+            },
+            options,
+            options: {
+                distance = 16,
+                positioner
+            }
+        } = tooltip;
+
+        const boundaries = {
+            left: scrollablePixelsX ? plotLeft : 0,
+            right: scrollablePixelsX ?
+                plotLeft + plotWidth - scrollablePixelsX : chartWidth,
+            top: scrollablePixelsY ? plotTop : 0,
+            bottom: scrollablePixelsY ?
+                plotTop + plotHeight - scrollablePixelsY : chartHeight
+        };
+
+        const tooltipLabel = tooltip.getLabel();
+        const headerTop = Boolean(opposite);
+
+        let distributionBoxTop = plotTop;
+        let headerHeight = 0;
+        let rightAlign = false;
+
+        /**
+         * Calculates the anchor position for the tooltip
```

Comment:
A bit confusing description - we are talking about a partial tooltip here, right? So the comment would be clearer if it read something like "Calculates the anchor position for the partial tooltip"